src/Common/src/Interop/Unix/System.Security.Cryptography.Native/Interop.Ssl.cs

```diff
@@ -283,9 +282,9 @@ protected override void Dispose(bool disposing)
                 _writeBio?.Dispose();
             }
 
-            if (_alpnHandle.IsAllocated)
+            if (AlpnHandle.IsAllocated)
             {
-                _alpnHandle.Free();
+                AlpnHandle.Free();
```

Comment:
Is this change necessary? We explicitly avoided doing this previously because get_AlpnHandle returns a copy, such that subsequent checks of AlpnHandle.IsAllocated will return true even though it's been freed.  By targeting _alpnHandle directly, we're targeting the original, not a copy, when calling Free.

Reply:
Without this change alpnhandle is not accessible outside this class. The handle value is required in SslStreamPal.Unix.cs, to verify if the target == null. Should this be a field?

Reply:
Could always ref return it
```csharp
private GCHandle _alpnHandle;
public ref GCHandle AlpnHandle => ref _alpnHandle;
```

Reply:
Can put this back to a set only and free inside the matching algo